app/lib/meetup.rb

```diff
@@ -0,0 +1,96 @@
+class Meetup
+  include HTTParty
+  base_uri 'api.meetup.com'
+
+  attr_reader :options
+
+  def initialize
+    api_key  = OperationCode.fetch_secret_with(name: :meetup_api_key)
+    @options = {
+      query: {
+      key: api_key,
+      sign: "true"
+      }
+    }
+  end
+
+  # GET all Operation Code Pro data. Returns all op-code groups.
+  def operationcode_data
+    response = self.class.get("/pro/operationcode/groups", options)
+    return_value_for response
+  end
+
+  # GET events for each group
+  def get_events_for(urlname)
+    response = self.class.get("/#{urlname}/events", options)
+    return_value_for response
+  end
+
+  #Return all meetups based on group names if venue is present. Return array of all op-code events
+  def event_details_by_group
+    events = []
+    group_names.each do |group|
+      get_events_for(group).each do |event|
+        next unless event['venue'].present?
+        events << details_for(event)
+      end
+    end
+    events.flatten
+  end
+
+  def add_events_to_database!
+    event_details_by_group.each do |event|
+      saved_event = Event.find_or_initialize_by(source_id: event[:source_id], source_type: "Meetup")
+      if saved_event.new_record? || saved_event_updated?(saved_event, event)
+        saved_event.update!(event)
+      end
+    end
+  end
+
+  def saved_event_updated?(saved_event, event)
```

Comment:
If this method is not being called outside of this class, we can move it down in the file and make it a `private` method.